levels.go

```diff
@@ -401,12 +404,18 @@ func (s *levelsController) compactBuildTables(level int, cd compactDef, limiter
 						continue // Skip adding this key.
 					}
 				}
-			}
-			if filter != nil {
-				skip := filter(it.Key(), vs.Value, vs.UserMeta)
-				if skip {
-					discardStats.collect(vs)
-					continue
+				if filter != nil {
+					switch filter.Filter(it.Key(), vs.Value, vs.UserMeta) {
+					case DecisionDelete:
+						// Convert to delete tombstone.
+						discardStats.collect(it.Value())
+						builder.Add(it.Key(), y.ValueStruct{Meta: bitDelete})
```

Comment:
What about set `skipKey` here? So we can delete the data of old version in next loop.
And it looks like we can simply fall back to `DecisionDrop` when the target level is the last level.